internal/entities/payments/transactions/core.go

```diff
@@ -28,16 +28,16 @@ func InitiateTransaction(){
 					}
 					payment.DestinationId=Wallet.ID
 				}
-				err=AddTransactions(payment,"from razorpay account")
+				err=AddTransactions(payment,constants.RzpWalletID,payment.DestinationId)
 				if err != nil {
 					payment.Status="failed"
 				}else{
 					payment.Status="processed"
 				}
 			}else{
-				err = AddTransactions(payment, "to razorpay account")
-				err2 = AddTransactions(payment,"from razorpay account")
-				if err != nil || err2!=nil{
+				if err = AddTransactions(payment, payment.SourceId,constants.RzpWalletID);err !=nil{
+					payment.Status="failed"
+				}else if err = AddTransactions(payment,constants.RzpWalletID,payment.DestinationId);err !=nil{
 					payment.Status="failed"
 				}else{
 					payment.Status="processed"
```

Comment:
return not written line 41
here payment should be added to channel after setting the status to "failed"/"processed"
even if the status is "failed" we add it to channel and update the payment in DB